lib/Diagnostics/Diagnostics.js

```diff
@@ -2,23 +2,62 @@
  * Created by gadyaari on 25/09/2016.
  */
 
-var config = require('../../common/Configuration');
-var loggerModule = require('../../common/logger');
-var ErrorUtils = require('./../utils/error-utils');
+const config = require('../../common/Configuration');
+const loggerModule = require('../../common/logger');
+const ErrorUtils = require('./../utils/error-utils');
+const BackendClient = require('../BackendClientFactory.js').getBackendClient();
+const DiagnosticsAlertsAnalyzer = require('./DiagnosticsAlertsAnalyzer');
+const diagAlertsErrorCodes = require('./DiagnosticsAlerts').diagnosticsErrorCodes;
+const _ = require('underscore');
+const Q = require('q');
+const EventEmitter = require('events');
 
-class Diagnostics {
-    constructor(loggerInfo) {
+/********************************************************************/
+/* Diagnostics holds two types of data structures:
+    1) An object called StaticMetaData combined of a data object and
+    a boolean flag for the class to know if data needs to be updated
+    to server.
+    StaticMetaData = {
+        modified: false
+        data: {},
+    }
+    2) An object called DynamicMetaData containing two arrays.
+    Dynamic object is sent to server regularly
+    DynamicMetaData = {
+        inputs: [],
+        flavors: []
+    }
+ */
+/********************************************************************/
+
+class Diagnostics extends EventEmitter {
+    constructor(entryId) {
+        super();
+        this._logger = loggerModule.getLogger("Diagnostics", `[${entryId}] `);
+        this._reportIntervalTimeInMSec = config.get('diagnostics').diagnosticsReportIntervalRunTimeInSec *1000;
         this.alertWindowSizeInMSec = config.get('diagnostics').errorsWindowSizeInMin * 60 * 1000;
+        this.staticMetaData = { modified: false };
+        this.dynamicMetaData = { inputs: [], flavors: [] };
+        this._diagnosticsAnalyzer = this._createAlertsAnalyzer.call(this, entryId);
         this._internalAlerts = [];
         this._externalAlerts = new Map();
-        this._logger = loggerModule.getLogger("Diagnostics", `[${loggerInfo}] `);
-        this.clearOldAlerts.call(this);
+        this._runDiagnosticsReport.call(this);
     }
 
     toJSON() {
         return { "internalAlerts" : this._internalAlerts, "externalAlerts" : Array.from(this._externalAlerts) };
     }
-    
+
+    _createAlertsAnalyzer(entryId) {
+        let an = new DiagnosticsAlertsAnalyzer(entryId, this.alertWindowSizeInMSec);
```

Comment:
you need to put DiagnosticsAlertsAnalyzer instance as a class member, otherwise GC will delete this object...

this.an=new DiagnosticsAlertsAnalyzer(entryId, this.alertWindowSizeInMSec);